src/components/learner-credit-management/search/CatalogSearch.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import algoliasearch from 'algoliasearch/lite';
+import { Configure, InstantSearch } from 'react-instantsearch-dom';
+
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { getConfig } from '@edx/frontend-platform/config';
+import { SearchHeader } from '@edx/frontend-enterprise-catalog-search';
+
+import { configuration } from '../../../config';
+import CatalogSearchResults from './CatalogSearchResults';
+
+const CatalogSearch = ({ offerId }) => {
+  const searchClient = algoliasearch(configuration.ALGOLIA.APP_ID, configuration.ALGOLIA.SEARCH_API_KEY);
+
+  const config = getConfig();
+  const courseIndex = useMemo(() => {
+    const cIndex = searchClient.initIndex(config.ALGOLIA_INDEX_NAME);
+    return cIndex;
+  }, [config.ALGOLIA_INDEX_NAME, searchClient]);
+
+  const searchFilters = `enterprise_catalog_query_uuids:${offerId}`;
```

Comment:
Just wanted to confirm that this is the correct filter, only allowing admins to see what is associated with the offerId fetched from the data analytics api.

Reply:
@kiram15 The Algolia filter should be based on a **catalog uuid**, not a **catalog query uuid**. The frontend isn't aware of catalog query uuids, only catalog uuids.

As previously discussed, we unfortunately don't have access to the catalog UUID for this page currently as the analytics API (like you noted earlier) only returns `catalog_name`, not `catalog_uuid`. We will be updating the data source for the LCM feature to no longer pull from the analytics API, in favor of pulling directly from enterprise-access/enterprise-subsidy. Until then, though, we won't have access to a catalog uuid to use in the search filter.

For the short term, we could either temporarily hardcode a catalog UUID until the data is available to the UI (behind a feature flag so not really a concern other than not being able to adequately test your changes on stage/prod at the moment) _OR_ (probably preferable) perhaps tap into the `enterprise_customer_catalogs` returned by the `https://courses.edx.org/enterprise/api/v1/enterprise-customer/dashboard_list/` API (the source of the enterprise customer metadata for admin portal).

Happy to sync up about short-term options in Slack!